CSLabs.Api/Controllers/ModuleController.cs

```diff
@@ -27,20 +27,30 @@ public class ModuleController : BaseController
         public ModuleController(BaseControllerDependencies deps) : base(deps)
         {
         }
-        // GET api/values
+        
+        // GET api/v1/Modules/?tagNames
         [HttpGet]
-        public IActionResult Get()
+        public async Task<IActionResult> Get(string tagNames = "")
         {
-            return Ok(DatabaseContext.Modules.Where(m => m.Published).ToList());
+            if (string.IsNullOrEmpty(tagNames))
+                return Ok(DatabaseContext.Modules.Where(m => m.Published).ToList());
+            var tagNamesList = tagNames.Split(",").Select(s => s.Trim()).ToList();
+            return Ok(
+                await DatabaseContext.Modules
+                    .Where(m => m.ModuleTags.Any(mt => tagNamesList.Contains(mt.Tag.Name)))
```

Comment:
Ahh we forgot to filter out unpublished modules.

```suggestion
                    .Where(m => m.Published)
                    .Where(m => m.ModuleTags.Any(mt => tagNamesList.Contains(mt.Tag.Name)))
```